### ISSUE TITLE
error in process filter; don't know how to interpret display ... org.macosforge.xquartz

### ISSUE THREAD

**Issue by csdrane** [stsquad/emacs_chrome#103]
I receive this error any time I try to use the Edit in Emacs plugin.

```
error in process filter: make-frame: Don't know how to interpret display ""/tmp/launch-IcrLfZ/org.macosforge.xquartz:0""
error in process filter: Don't know how to interpret display ""/tmp/launch-IcrLfZ/org.macosforge.xquartz:0""
```

**Comment by alpha22jp** [comment]
Hi @stsquad

I'm the author of [Atomic Chrome for Emacs](https://github.com/alpha22jp/atomic-chrome) which is similar to Edit with Emacs. This package is very helpful for me to know how to open a new frame and get it focused.

BTW, It appears that the last commit (0a50fbb) breaks the behaviour of opening a new frame. When I start Emacs with `emacs --daemon` and press the edit button on Chrome, a new frame doesn't popup. I think it's because `window-system` returns `nil` while Emacs has no frame.

Am I misunerstanding something? I'm using Emacs 25.1 on Linux (ubuntu 14.04).

**Comment by stsquad** [comment]
@alpha22jp yeah I think the logic is a little broken. Fundamentally (window-system) reports the current frames display so is a poor test for deciding if we can create a new graphical frame.

**Comment by stsquad** [comment]
Well it works if there was a graphical window at some point, which is my default, I only enable edit-servers when:

    (when (and (getenv "DISPLAY") (daemonp) (not I-am-root))
      (if (locate-library "atomic-chrome")
          (use-package my-atomic-chrome)
        (use-package my-edit-server)))

The failure only occurs if I start an emacs daemon session in a terminal session without display set. However if I create a X frame via emacsclient things work after that. I think we need to delve into the display code to see how emacsclient knows if it can display something on X.

**Comment by stsquad** [comment]
OK this fix has caused #132 and #133 so we need a decent solution that keeps both use cases happy. It seems the --daemon does not keep window-system set.
